src/Compilers/Core/Portable/SpecialTypeExtensions.cs

```diff
@@ -57,6 +57,35 @@ public static bool IsBlittable(this SpecialType specialType)
             }
         }
 
+        /// <summary>
+        /// Checks if a type is an immutable value type primitive.
+        /// </summary>
+        public static bool IsImmutableValueTypePrimitive(this SpecialType specialType)
```

Comment:
Agree with @sharwell 

Better to find a place down the road where we forgot to do this optimization vs. finding out we accidentally started doing it to a type where it is not safe.